install/kubernetes/cilium/templates/clustermesh-apiserver/deployment.yaml

```diff
@@ -204,11 +205,19 @@ spec:
               name: cilium-config
               key: enable-k8s-endpoint-slice
               optional: true
+        readinessProbe:
+          httpGet:
+            path: /readyz
+            port: health
+          periodSeconds: 1
```

Comment:
I personally agree that one second seems aggressive, especially because the current implementation will never turn unready after initialization. 

Technically speaking, a `startupProbe` looks more appropriate in this context (quoting the description `StartupProbe indicates that the Pod has successfully initialized.`. The main issue, though, is that it is tricky to pick the correct failure threshold (which would also need to be user configurable), because it would eventually cause the restart of the pod. Yet, in large scale clusters, it is expected that the clustermesh-apiserver takes a quite significant time during initial synchronization, due to etcd rate limiting. Which makes me wonder whether we should either set extremely high values (in the order of hours) or just stick to the readiness probe.

Reply:
Good point, this probe will continue to run even after it succeeds - so 1s is probably too aggressive. I considered using a startupProbe, but as mentioned, the startup time can vary greatly and picking the right threshold could be tricky. I didn't add locking because the fields are only written to once after the sync is complete, and the request handler is only reading. If there was a race it would cause 1 extra probe before being marked ready, which with a 1s threshold seems pretty negligible. But with an increased threshold its not, and is just better design anyway. I'll update with these changes